src/core/preprocessor.js

```diff
@@ -0,0 +1,256 @@
+import { Debug } from './debug.js';
+
+// set to true to enable debug logging
+const enableLogging = false;
```

Comment:
I love this idea, really useful. It does mean the log calls in this file remain in production code though.

Perhaps we could embrace a more general approach to trace logging?

We could add `Debug.trace` function which takes a named "stream" parameter and allow user to enable/disable streams at runtime. Calls to `Debug.trace` (and the stream enable calls) would be removed by rollup as usual.

Reply:
Yep I thought about it few times as well .. and now is a good time to do it. I'll add description to this PR.